tests/zkevm/test_worst_compute.py

```diff
@@ -237,23 +243,173 @@ def test_worst_modexp(
 
 
 @pytest.mark.valid_from("Cancun")
-def test_worst_ecrecover(
+@pytest.mark.parametrize(
+    "precompile_address,parameters",
+    [
+        pytest.param(
+            0x01,
+            [
+                "38D18ACB67D25C8BB9942764B62F18E17054F66A817BD4295423ADF9ED98873E",
+                "000000000000000000000000000000000000000000000000000000000000001B",
+                "38D18ACB67D25C8BB9942764B62F18E17054F66A817BD4295423ADF9ED98873E",
+                "789D1DD423D25F0772D2748D60F7E4B81BB14D086EBA8E8E8EFB6DCFF8A4AE02",
+            ],
+            id="ecrecover",
+        ),
+        pytest.param(
+            0x06,
+            [
+                "18B18ACFB4C2C30276DB5411368E7185B311DD124691610C5D3B74034E093DC9",
+                "063C909C4720840CB5134CB9F59FA749755796819658D32EFC0D288198F37266",
+                "07C2B7F58A84BD6145F00C9C2BC0BB1A187F20FF2C92963A88019E7C6A014EED",
+                "06614E20C147E940F2D70DA3F74C9A17DF361706A4485C742BD6788478FA17D7",
+            ],
+            id="bn128_add",
+        ),
+        pytest.param(
+            0x07,
+            [
+                "1A87B0584CE92F4593D161480614F2989035225609F08058CCFA3D0F940FEBE3",
+                "1A2F3C951F6DADCC7EE9007DFF81504B0FCD6D7CF59996EFDC33D92BF7F9F8F6",
+                "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
+            ],
+            id="bn128_mul",
+        ),
+        pytest.param(
+            0x08,
+            [
+                # First pairing
```

Comment:
Have you check what is the worst?
- 2 pairings: result is 1, no exp amortization,
- 1 pairing: result is 0, even less exp amortization,
- a lot of pairings: exp amortization.

Reply:
It is not, but mainly because for the full amortization we need a separate test than this generic one to deal with the total input size not to blow up the max contract size coming from PUSHes for MSTORE. Probably using CALLDATA.

I'll add a TODO that this case can be iterated further. As an extra dimension to improve might be particular pairing cases that could be worse.